cql3/functions/function.hh

```diff
@@ -50,6 +50,13 @@ public:
      */
     virtual bool is_aggregate() const = 0;
 
+    /**
+     * Checks wheather the function can be distributed and is able to reduce states.
+     *
+     * @return <code>true</code> if the function is reducible, <code>false</code> otherwise.
+     */
+    virtual bool is_reducible() const = 0;
+
```

Comment:
Better to move to aggregate, no?

Reply:
Do you mean to `aggregate_function` or to `aggregate`?

I've followed `is_aggregate()` function which is defined at `function` level, not `aggregate_function`. The difference would be in some places this would require to cast `function` to `aggregate_function` and then check if it is reducible.

Reply:
I can't think of why we'd need to check for reducibility without knowing it's an aggregate. But if you keep it here, rename it to is_reducible_aggregate() to indicate it's not generic for all functions.

Reply:
Moved to `aggregate_function`